dex/pair/src/fee.rs

```diff
@@ -51,25 +52,24 @@ pub trait FeeModule:
         pair_address: ManagedAddress,
         first_token: TokenIdentifier,
         second_token: TokenIdentifier,
-    ) -> SCResult<()> {
```

Comment:
For methods that are deeply nested and only do very basic things, removing the `SCResult<()>` to not propagate it all over is fine. I don't think we should remove `SCResult` from _everywhere_ though, as that makes the code less clear and it doesn't really bring any improvement. I'd say keep endpoints that do checks themselves return SCResult<()> instead of using `assert!` everyhwere.

This also means keeping high-level check methods like `require_permissions` with `SCResult` return.

Reply:
Imo it's pretty straight forward what require_permissions does without returning a SCResult. Assert is just fine. Think that's why Rust guys implemented it